PreferredUILang tests fail on Windows10

Test 'Microsoft.CodeAnalysis.CSharp.CommandLine.UnitTests.CommandLineTests.PreferredUILang' failed: Assert.Contains() failure:
Not found: CS2038
In value: Microsoft (R) Visual C# Compiler version 42.42.42.42
Copyright (C) Microsoft Corporation. All rights reserved.
warning CS2008: No source files specified.
error CS1562: Outputs without source must have the /out option specified
C:\R1\Open\src\Compilers\CSharp\Test\CommandLine\CommandLineTests.cs(6258,0): at Microsoft.CodeAnalysis.CSharp.CommandLine.UnitTests.CommandLineTests.PreferredUILang()

--- a/src/Compilers/CSharp/Desktop/CommandLine/CommandLineParser.cs
+++ b/src/Compilers/CSharp/Desktop/CommandLine/CommandLineParser.cs
@@ -262,11 +262,21 @@ namespace Microsoft.CodeAnalysis.CSharp
                         try
                         {
                             preferredUILang = new CultureInfo(value);
+                            if ((preferredUILang.CultureTypes & CultureTypes.UserCustomCulture) != 0)
+                            {
+                                // Do not use user custom cultures.
+                                preferredUILang = null;
+                            }
                         }
                         catch (CultureNotFoundException)
                         {
+                        }
+
+                        if (preferredUILang == null)
+                        {
                             AddDiagnostic(diagnostics, ErrorCode.WRN_BadUILang, value);
                         }
+
                         continue;
 
 #if DEBUG

--- a/src/Compilers/VisualBasic/Desktop/CommandLine/CommandLineParser.vb
+++ b/src/Compilers/VisualBasic/Desktop/CommandLine/CommandLineParser.vb
@@ -356,9 +356,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
                         Try
                             preferredUILang = New CultureInfo(value)
+                            If (preferredUILang.CultureTypes And CultureTypes.UserCustomCulture) <> 0 Then
+                                ' Do not use user custom cultures.
+                                preferredUILang = Nothing
+                            End If
                         Catch ex As CultureNotFoundException
-                            AddDiagnostic(diagnostics, ERRID.WRN_BadUILang, value)
                         End Try
+
+                        If preferredUILang Is Nothing Then
+                            AddDiagnostic(diagnostics, ERRID.WRN_BadUILang, value)
+                        End If
+
                         Continue For
 #If DEBUG Then
                     Case "attachdebugger"

